apps/testingaccessibility/src/pages/api/prices.ts

```diff
@@ -50,13 +50,21 @@ const pricesHandler = async (req: NextApiRequest, res: NextApiResponse) => {
             return upgradeProductIds.includes(purchase.productId)
           })?.id
 
-      const {activeMerchantCoupon, defaultCoupon} =
-        await getActiveMerchantCoupon({
-          siteCouponId,
-          code,
-          productId,
-          spanContext,
-        })
+      // explicit incoming merchant coupons are honored
+      // without checking for other potential coupons
+      // if there is no explicit incoming merchant coupon
+      // we check for default/global coupon or an incoming code
+      const {activeMerchantCoupon, defaultCoupon} = merchantCouponId
+        ? {
+            activeMerchantCoupon: {id: merchantCouponId},
+            defaultCoupon: false,
+          }
```

Comment:
if there is an explicitly applied coupon we can just skip the lookup and save some time on the response